core/events/events_viewport.ts

```diff
@@ -96,5 +96,12 @@ export class ViewportChange extends UiBase {
   }
 }
 
+export interface ViewportChangeJson extends AbstractEventJson {
+  viewTop: number;
```

Comment:
is it weird that if you create the event from js these are optional but not if you create it from json? what about allowing these to be optional but set them to 0 in `fromJson` just like in the constructor?

Reply:
I'm not sure that it's weird, because they're optional for different reasons. They're optional in the constructor so that you can create an event, and then call `fromJson` on it (which is a really ugly pattern, `fromJson` should be `static`, not an instance method).